### ISSUE TITLE
Feature request: New configuration for open file

### ISSUE THREAD

**Issue by hkroe** [pellepl/spiffs#161]
Would it be possible to add a configuration so that a file can only be opened once at a time?
At a first glance I would say that this isn’t a big change.

**Comment by pellepl** [comment]
Hi,
yes, I believe that would be easy enough. Being curious here, what is the use case? 

Just so I understood it correctly, you want a config flag preventing users opening any already opened file? Doing so would yield something like SPIFFS_ERR_FILE_ALREADY_OPENED or something?

**Comment by hkroe** [comment]
Hi
> Just so I understood it correctly, you want a config flag preventing users opening any already opened file? Doing so would yield something like SPIFFS_ERR_FILE_ALREADY_OPENED or something?

Yes, this is correct.

> Being curious here, what is the use case?

I just have doubts about race conditions.
For instance, one file was opened by two Tasks (Task A and Task B).
- Task A reads the file size.
- While Task A is calculating an offset Task B gets the processor time.
- Task B writes to the file -> Task A calculates with a wrong file size.

**Comment by valkuc** [comment]
Shouldn't you use threads synchronization techniques for that purpose. You could use mutex or signals.

**Comment by hkroe** [comment]
Yes, you should. But you can check for this new error code to be sure.
